cdap-watchdog/src/main/java/co/cask/cdap/logging/save/LogSaver.java

```diff
@@ -248,6 +125,10 @@ private void cancelLogCollectorCallbacks() {
   private void subscribe(Set<Integer> partitions) throws Exception {
     LOG.info("Prepare to subscribe for partitions: {}", partitions);
 
+    for (KafkaLogProcessor processor : messageProcessors) {
+      processor.init(partitions);
```

Comment:
Not fixing this, we should not process if there an exception on init for all plugins. Would give the admins a chance to look at the failure and not fail silently.